content/en/blog/_posts/2021-09-06-memory-qos-cgroups-v2/index.md

```diff
@@ -0,0 +1,85 @@
+---
+layout: blog
+title: 'Alpha in Kubernetes v1.22: Quality-of-Service for Memory Resources'
+date: 2021-09-01
+slug: qos-memory-resources
+---
+
+**Authors:** Tim Xu (Tencent Cloud)]
+
+Kubernetes v1.22 introduced a new alpha feature that improves how Linux nodes implement memory resource requests and limits.
+
+In prior releases, Kubernetes did not support memory quality guarantees. In current Kubernetes design, user sets requests and limits to describe  resources required for containers in pod, and Kubernetes uses these values for pod scheduling and creation. For resources such as CPU and memory, Kubernetes allocates and limits these through cgroup on Linux nodes for isolation purposes. CPU is well protected in existing implementations, but memory is not as well protected due to cgroup v1 limitations. Fortunately, cgroup v2 brings a new design and implementation to achieve full protection on memory. The new feature relies on cgroups v2 which most current operating system releases for Linux already provide. With this experimental feature, [quality-of-service for pods and containers](/docs/tasks/configure-pod-container/quality-service-pod/) extends to cover not just CPU time but memory as well.
+
+## How does it work?
+Memory QoS uses the memory controller of cgroup v2 to guarantee memory resources in Kubernetes. Memory requests and limits of containers in pod are used to set specific interfaces `memory.min` and `memory.high` provided by the memory controller. When `memory.min` is set to memory requests, memory resources are reserved and never reclaimed by the kernel; this is how Memory QoS ensures the availability of memory for Kubernetes pods. And if memory limits are set in the container, this means that the system needs to limit container memory usage, Memory Qos uses `memory.high` to do the throttle, ensuring that the system is not overwhelmed by instantaneous memory allocation.
```

Comment:
slight reword?

s/do the throttle/throttle workload approaching it's memory limit, ...

Reply:
done